plugins/reporters/spdx/src/main/kotlin/Extensions.kt

```diff
@@ -181,10 +181,7 @@ internal fun Package.toSpdxPackage(
             licenseInfoResolver.resolveLicenseInfo(id)
                 .filterExcluded()
                 .filter(LicenseView.ONLY_DETECTED)
-                .map { resolvedLicense ->
-                    resolvedLicense.license.takeIf { it.isValid(SpdxExpression.Strictness.ALLOW_DEPRECATED) }
-                        .nullOrBlankToSpdxNoassertionOrNone()
-                }
+                .map { it.license.nullOrBlankToSpdxNoassertionOrNone() }
```

Comment:
> The rationale is already described

Could be, maybe I understand it wrong...

If you have an ORT `Package` which has deprecated SPDX license IDs,
and you call `toSpdxPackage()` in it then:

1. before: Deprecated IDs get filtered out, check in constructor suceeds.
2. after: Deprecated IDs do not get filtered out, check in constructor fails. Exception is thrown.

Does this make any sense?

Reply:
IMO it's different: When using a deprecated license ID, then `toSpdxPackage()`

- before, would *not* filter it out, but keep it due to `ALLOW_DEPRECATED`, and then `SpdxPackage`'s constructor would throw as it uses `ALLOW_LICENSEREF_EXCEPTIONS` (which does *not* allow deprecated IDs).
- after, straight `SpdxPackage`'s constructor would throw, so no change in behavior.

Reply:
Indeed. You're right.

Let's consider "LicenseRef" exception behavior as well. Is the following understanding correct?exceptions"?

1. before: get filtered out
2. after: get kept

Reply:
Yes, that's correct.

Unfortunately, once more the SPDX specification is vague as it does not clearly say whether exceptions (`LicenseRef` ones or not) should be included in `licenseInfoFromFiles`. It just says to not specify *relationships* and mentions "i.e., conjunctive, disjunctive", so I would assume the `WITH` operator is not meant to be a "relationship" and exceptions should be included. And if exceptions are included, I'd also allow `LicenseRef` exceptions as otherwise we'll again run into issue #9111.

Reply:
I'm not 100%, but IIRC this validation [1] flags `LicenseRef-` as exceptions.

[1] https://github.com/spdx/tools

Reply:
> as that was the intention of [aa3959e](https://github.com/oss-review-toolkit/ort/commit/aa3959e6eba78383681af6a45bc3e9d31e03d853).

That PR also contained 4c6c411dc2f24e532411636e7c1d204170df28a3, which is the bit I had tried but didn't manage to find. I believe your change is inline with both of these commits. So, I'm not worried anymore. However,
regarding

> What's the rationale for no more filtering out license ref exceptions here?

I believe the rationale in the commit message can be improved. E.g. say somehow that
the intention was to violated the SPDX spec in this regard intentionally, but log warnings for the violations - and reference both of these commits. 

Does this make sense to you too?

Reply:
Not sure what exactly you had in mind, but I've added a paragraph to the commit message that hopefully matches your intention.

Reply:
Thanks!